File: backend/auth2/src/main/java/com/ewatts/auth2/dto/UserDto.java
package com.ewatts.auth2.dto;

import com.ewatts.auth2.entity.AuthProvider;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private Long id;
    private String username;
    private String email;
    private String phoneNumber;
    private boolean emailVerified;
    private AuthProvider provider;
    private LocalDateTime createdAt;
    private String profilePictureUrl;
}

File: my_app/lib/models/auth/user_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_dto.g.dart';

enum AuthProvider { LOCAL, GOOGLE, FACEBOOK }

@JsonSerializable()
class UserDto {
  final int id;
  final String username;
  final String email;
  final String? phoneNumber;
  final bool emailVerified;
  final AuthProvider provider;
  final DateTime createdAt;
  final String? profilePictureUrl;

  UserDto({
    required this.id,
    required this.username,
    required this.email,
    this.phoneNumber,
    required this.emailVerified,
    required this.provider,
    required this.createdAt,
    this.profilePictureUrl,
  });

  factory UserDto.fromJson(Map<String, dynamic> json) =>
      _$UserDtoFromJson(json);
  Map<String, dynamic> toJson() => _$UserDtoToJson(this);
}
